lib/exq/redis/script.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Exq.Redis.Script do
+  alias Exq.Redis.Connection
+
+  defmodule Prepare do
+    def script(source) do
+      hash = :crypto.hash(:sha, source)
+                    |> Base.encode16(case: :lower)
+
+      {hash, source}
+    end
+  end
+
+  @scripts %{
+    schedule: Prepare.script("""
+      local schedule_queue_key, job_queue_key = KEYS[1], KEYS[2]
+      local jid, score, job_serialized = ARGV[1], ARGV[2], ARGV[3]
+
+      redis.call('ZADD', schedule_queue_key, score, jid)
+      redis.call('HSET', schedule_queue_key .. ':job', jid, job_serialized)
+      redis.call('HSET', schedule_queue_key .. ':job_queue', jid, job_queue_key)
+      """),
+    schedule_dequeue: Prepare.script("""
+      local schedule_queue_key, queues_key = KEYS[1], KEYS[2]
+      local score, page_size = ARGV[1], ARGV[2]
+
+      local dequeue_count = 0
+      for i, jid in ipairs(redis.call('ZRANGEBYSCORE', schedule_queue_key, 0, score, 'LIMIT', 0, page_size)) do
+        local job_serialized = redis.call('HGET', schedule_queue_key .. ':job', jid)
+        local job_queue_key = redis.call('HGET', schedule_queue_key .. ':job_queue', jid)
+
+        redis.call('SADD', 'queues', queues_key, job_queue_key)
+        redis.call('LPUSH', job_queue_key, job_serialized)
+
+        redis.call('ZREM', schedule_queue_key, jid)
+        redis.call('HDEL', schedule_queue_key .. ':job', jid)
+        redis.call('HDEL', schedule_queue_key .. ':job_queue', jid)
```

Comment:
It looks like the data structure used to store the schedules has changed. How would the upgrade work for existing users? I am assuming the changes are required because queue_name is not available in lua context. Is it possible to parse the job using [cjson](https://redis.io/commands/eval#cjson) or that would be too slow?